src/test/shell/bazel/bazel_symlink_test.sh

```diff
@@ -716,4 +721,179 @@ EOF
   bazel build --experimental_allow_unresolved_symlinks //a:exec || fail "build failed"
 }
 
+function setup_unresolved_symlink_as_input() {
+  add_to_bazelrc build --experimental_allow_unresolved_symlinks
+
+  mkdir -p pkg
+  cat > pkg/def.bzl <<'EOF'
+def _r(ctx):
+  symlink = ctx.actions.declare_symlink(ctx.label.name + "_s")
+  ctx.actions.symlink(output = symlink, target_path = ctx.file.file.basename)
```

Comment:
nit: why not make this an actual dangling symlink to better exercise the machinery of Bazel that does *not* resolve them when they are on the input of another action?

Reply:
Whether or not the symlink is dangling here is configured via the `stage_target` attribute. The actual tests using this setup function go over all three cases (file not even built, file built but not declared as an input, file declared as an input). Do you want me to change this?

Reply:
Oh, I see, now I understand why you did this. I thought that it would be better if the fact that the symlink is dangling was obvious from its target, but that would make it impossible to test all three cases so neatly. I'd probably have run one single action with the three kinds of inputs, but I don't think it's worth your time to change to that approach; I don't have an actual argument, it's just my taste, so meh.